app/scripts/lib/derived-keys.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
Indeed.  FWIW, I hand-generated the test vectors using a python HKDF lib as a rudimentary cross-check.